pkg/apiserver/handlers/endpoint/handler.go

```diff
@@ -18,33 +18,85 @@ import (
 	"encoding/json"
 	"net/http"
 
+	"antrea.io/antrea/pkg/apis/controlplane"
 	"antrea.io/antrea/pkg/controller/networkpolicy"
 )
 
+// EndpointQueryResponse is the reply struct for anctl endpoint queries
+type EndpointQueryResponse struct {
+	Endpoints []Endpoint `json:"endpoints,omitempty"`
+}
+
+type Rule struct {
+	PolicyRef controlplane.NetworkPolicyReference
+	Direction controlplane.Direction
+	RuleIndex int
+}
+
+type Endpoint struct {
+	Namespace string                                `json:"namespace,omitempty"`
+	Name      string                                `json:"name,omitempty"`
+	Policies  []controlplane.NetworkPolicyReference `json:"policies,omitempty"`
+	Rules     []Rule                                `json:"rules,omitempty"`
+}
+
 // HandleFunc creates a http.HandlerFunc which uses an AgentNetworkPolicyInfoQuerier
 // to query network policy rules in current agent.
 func HandleFunc(eq networkpolicy.EndpointQuerier) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
 		podName := r.URL.Query().Get("pod")
 		namespace := r.URL.Query().Get("namespace")
-		if namespace == "" {
-			namespace = "default"
-		}
 		// check for incomplete arguments
 		if podName == "" {
 			http.Error(w, "pod must be provided", http.StatusBadRequest)
 			return
 		}
 		// query endpoint and handle response errors
-		endpointQueryResponse, err := eq.QueryNetworkPolicies(namespace, podName)
+		endpointNetworkPolicyRules, err := eq.QueryNetworkPolicyRules(namespace, podName)
 		if err != nil {
 			http.Error(w, err.Error(), http.StatusInternalServerError)
 			return
 		}
-		if endpointQueryResponse == nil {
+		if endpointNetworkPolicyRules == nil {
 			http.Error(w, "could not find any endpoints matching your selection", http.StatusNotFound)
 			return
 		}
+
+		// make response policies
+		responsePolicies := make([]controlplane.NetworkPolicyReference, 0)
+		for _, internalPolicy := range endpointNetworkPolicyRules.AppliedPolicies {
+			responsePolicy := controlplane.NetworkPolicyReference{
+				Type:      internalPolicy.SourceRef.Type,
```

Comment:
Isn't the `SourceRef` already type `controlplane.NetworkPolicyReference`?

Reply:
Missed this, updated.